runtime/include/chpl-cache.h

```diff
@@ -37,6 +37,7 @@ extern "C" {
 
 // Is the cache supposed to be enabled? (set at compile time)
 extern const int CHPL_CACHE_REMOTE;
+extern c_nodeid_t chpl_nodeID;
```

Comment:
I'd be a lot happier without this. From a brief search it looks like it's only used in `chpl_cache_warn_if_disabled()`, which is in turn only called from `chpl_cache_init()` in `chpl-cache.c`. Could we simply move `chpl_cache_warn_if_disabled()` into `chpl-cache.c` therefore, and remove this `extern` decl?

Reply:
Not quite that simple since `chpl_cache_warn_if_disabled()` uses macros that undef'd in the header, but I can move just the `chpl_nodeID` check if that works for you.

Reply:
Feeling confused why this was needed in this PR at all --  the only mention of `chpl_nodeID` in the diff is for `chpl_msg(2, "%d: enter barrier for '%s'\n", chpl_nodeID, msg);` which obviously we could implement in a .c file somewhere, right?

Reply:
Above I changed a `chpl-comm.h` include to just include `chpl-comm-task-decls.h` to avoid a circular import between `chpl-cache.h` and `chpl-comm.h`

Reply:
Moving the `chpl_nodeID` check would be fine with me, yeah.

Reply:
Moved chpl_nodeID